tfexec/internal/e2etest/show_test.go

```diff
@@ -111,16 +111,15 @@ func TestShow_emptyDir(t *testing.T) {
 }
 
 func TestShow_noInitBasic(t *testing.T) {
-	// Prior to v1.2.0, running show before init always results in ErrNoInit.
+	// Prior to v1.2.0, running show before init always results in an error.
 	// In the basic case, in which the local backend is implicit and there are
 	// no providers to download, this is unintended behaviour, as
 	// init is not actually necessary. This is considered a known issue in
 	// pre-1.2.0 versions.
 	runTestVersions(t, []string{testutil.Latest012, testutil.Latest013, testutil.Latest014, testutil.Latest015, testutil.Latest_v1, testutil.Latest_v1_1}, "basic", func(t *testing.T, tfv *version.Version, tf *tfexec.Terraform) {
-		var noInit *tfexec.ErrNoInit
 		_, err := tf.Show(context.Background())
-		if !errors.As(err, &noInit) {
-			t.Fatalf("expected error ErrNoInit, got %T: %s", err, err)
+		if err == nil {
```

Comment:
Nit: Not sure if its worth checking for the (new) expected error in these cases or not.